app/views/users/_form.html.erb

```diff
@@ -0,0 +1,21 @@
+<% if @user.errors.any? %> 
+  <h2>The following errors prevented the user from being saved</h2> 
+  <ul> 
+    <% @user.errors.full_messages.each do |msg| %> 
+      <li><%= msg %></li> 
+    <% end %> 
+  </ul>
+<% end %>
```

Comment:
**Optional**
Move this logic to a helper. There should be absolutely no logic in the view. Views are used for displaying data that are passed to them from the controller and not for generating data.